internal/manager/controllerdef.go

```diff
@@ -217,6 +220,27 @@ func setupControllers(mgr manager.Manager, proxy proxy.Proxy, c *Config, feature
 				IngressClassName: c.IngressClassName,
 			},
 		},
+		// ---------------------------------------------------------------------------
+		// GatewayAPI Controllers
+		// ---------------------------------------------------------------------------
+		{
+			Enabled: featureGates["Gateway"],
+			AutoHandler: crdExistsChecker{
+				GVR: schema.GroupVersionResource{
+					Group:    gatewayv1alpha2.SchemeGroupVersion.Group,
+					Version:  gatewayv1alpha2.SchemeGroupVersion.Version,
+					Resource: "gateways",
+				}}.CRDExists,
+			Controller: &gateway.GatewayReconciler{
+				Client:          mgr.GetClient(),
+				Log:             ctrl.Log.WithName("controllers").WithName("Gateway"),
+				Scheme:          mgr.GetScheme(),
+				Proxy:           proxy,
+				PublishService:  c.PublishService,
+				KongConfig:      kongConfig,
```

Comment:
IMO including the whole of Kongconfig is too broad and hurts cohesion. I suggest injecting just a listener getter function, or the list of listeners read just once upon startup.

Reply:
The `Gateway` resource is meant to be a living reflection of the Kong Gateway itself, and needs access to the data-plane in order to make that reflection work. I'm really not seeing how having controllers in our control plane which is entirely about this data-plane having access to that data-plane's API to be too broad or not cohesive. We do however usually wrap these kinds of things behind the `Proxy` interface which was purpose built to be such an abstraction, so I moved the logic under there in 4d0023108da415a2bc842794f5c8af63a63fbd55 let me know if that seems sufficient.

Reply:
Yes this looks reasonable :+1: